quickjs.c

```diff
@@ -50421,12 +50425,9 @@ static JSValue js_dataview_getValue(JSContext *ctx,
     size = 1 << typed_array_size_log2(class_id);
     if (JS_ToIndex(ctx, &pos, argv[0]))
         return JS_EXCEPTION;
-    is_swap = FALSE;
+    is_swap = TRUE;
     if (argc > 1)
-        is_swap = JS_ToBool(ctx, argv[1]);
-#ifndef WORDS_BIGENDIAN
-    is_swap ^= 1;
-#endif
+        is_swap = !JS_ToBool(ctx, argv[1]);
```

Comment:
Can you walk me through this? `view.getInt32(42)` should read a big-endian value, meaning:

1. Byte-swap when system is LE
2. No byte-swap when system is BE

And vice versa, `view.getInt32(42, /*little endian*/true)` means:

1. No byte-swap when system is LE
2. Byte-swap when system is BE

It seems to me the logic should be something like this?
```c
is_le = argc > 1 && JS_ToBool(ctx, argv[1]);
is_swap = is_le ^ !is_be();
```

Reply:
@bnoordhuis: good catch! you are correct, I shall post a new commit later today.